xbout/load.py

```diff
@@ -158,52 +156,50 @@ def _arrange_for_concatenation(filepaths, nxpe=1, nype=1):
     return paths_grid, concat_dims
 
 
-def _trim(ds, ghosts, guards={}, keep_guards={}, nxpe=1, nype=1):
+def _trim(ds, *, guards, keep_boundaries, nxpe, nype):
     """
-    Trims all ghost and guard cells off a single dataset read from a single
-    BOUT dump file, to prepare for concatenation.
+    Trims all guard (and optionally boundary) cells off a single dataset read from a
+    single BOUT dump file, to prepare for concatenation.
     Also drops some variables that store timing information, which are different for each
     process and so cannot be concatenated.
 
     Parameters
     ----------
-    ghosts : dict, optional
-        Number of ghost cells along each dimension, e.g. {'x': 2, 't': 0}
-    guards : dict, optional
-        Number of guard cells along each dimension, e.g. {'x': 2, 'y': 2}
-    keep_guards : dict, optional
-        Whether or not to preserve the guard cells along each dimension, e.g.
+    guards : dict
+        Number of guard cells along each dimension, e.g. {'x': 2, 't': 0}
+    keep_boundaries : dict
+        Whether or not to preserve the boundary cells along each dimension, e.g.
```

Comment:
Just for reference, `nxpe`, `nype` aren't documented here

Also, I guess `guards` and `keep_boundaries` are optional, although that is somewhat obvious from the signature (if you're familiar with the syntax!)

Reply:
I've added `nxpe` and `nype` to the docstring.

`guards` and `keep_boundaries` aren't optional. The `foo(*, a, b)` syntax means that `a` and `b` are required keyword arguments. I found it recently and quite like it because I think long sequences of positional arguments are bug-prone, but it's still nice to use required arguments sometimes - for example here it's an internal-implementation-only function which isn't called with default arguments, so no need to have the handling for default arguments (that would be needed if we had something like `guards=None`).